src/elfpy/pricing_models.py

```diff
@@ -824,6 +842,23 @@ def calc_in_given_out(
             f"\n\tspot_price={spot_price}\n\tk={k}\n\twithout_fee_or_slippage={without_fee_or_slippage}"
             f"\n\twithout_fee={without_fee}\n\tfee={fee}"
         )
+
+        # TODO(jalextowle): With some analysis, it seems possible to show that
+        # we skip straight from non-negative reals to the complex plane without
+        # hitting negative reals.
+        #
+        # Ensure that the outputs are all non-negative floats. We only need to
+        # check with_fee since without_fee_or_slippage will always be a positive
+        # float due to the constraints on the inputs, without_fee = with_fee + fee
```

Comment:
In this case, `without_fee = with_fee - fee`, so we should instead check that `without_fee` is a non-negative float. We can change the comment to indicate that we don't need to check `with_fee` since `with_fee = without_fee + fee`.

Reply:
sounds like we can remove it, don't see an upper-bound that's worth checking

Reply:
oh i see, we'll just check the lower of the two, before fees for non-negativity 👌